backend/Dockerfile

```diff
@@ -13,24 +28,25 @@ RUN aws configure set default.region ap-southeast-1
 
 WORKDIR /usr/home/postmangovsg
 
-COPY shared ../shared
-RUN cd ../shared && npm ci
+# this means npm will only install production deps, and optimizations in the node
+# engine will take place as well
+ENV NODE_ENV=production 
 
-COPY ./package* ./
-RUN npm ci
+# expose port and copy these files first as they're unlikely to change
+EXPOSE 4000
+COPY ./docker-entrypoint.sh ./
+RUN ["chmod", "+x", "docker-entrypoint.sh"]
 
-COPY tsconfig.json ./
-COPY tsconfig.build.json ./
-COPY openapi.yaml ./
-COPY src ./src
+COPY --from=builder /usr/home/shared/package* ../shared/
+RUN cd ../shared && npm install 
```

Comment:
Do we need npm install again if the folder is copied from the builder step ?

Reply:
Nope, we only copy package files